drivers/gpio/gpio_renesas_rz.c

```diff
@@ -0,0 +1,496 @@
+	fsp_err_t err;
+	ioport_size_t port_value;
+
+	err = config->fsp_api->portRead(data->fsp_ctrl, config->fsp_port, &port_value);
+	if (err != FSP_SUCCESS) {
+		return -EIO;
+	}
+	*value = (gpio_port_value_t)port_value;
+	return 0;
+}
+
+static int gpio_rz_port_set_masked_raw(const struct device *dev, gpio_port_pins_t mask,
+				       gpio_port_value_t value)
+{
+	const struct gpio_rz_config *config = dev->config;
+	struct gpio_rz_data *data = dev->data;
+	ioport_size_t port_mask = (ioport_size_t)mask;
+	ioport_size_t port_value = (ioport_size_t)value;
+	fsp_err_t err;
+
+	err = config->fsp_api->portWrite(data->fsp_ctrl, config->fsp_port, port_value, port_mask);
+	if (err != FSP_SUCCESS) {
+		return -EIO;
+	}
+	return 0;
+}
+
+static int gpio_rz_port_set_bits_raw(const struct device *dev, gpio_port_pins_t pins)
+{
+	const struct gpio_rz_config *config = dev->config;
+	struct gpio_rz_data *data = dev->data;
+	ioport_size_t mask = (ioport_size_t)pins;
+	ioport_size_t value = (ioport_size_t)pins;
+	fsp_err_t err;
+
+	err = config->fsp_api->portWrite(data->fsp_ctrl, config->fsp_port, value, mask);
+	if (err != FSP_SUCCESS) {
+		return -EIO;
+	}
+	return 0;
+}
+
+static int gpio_rz_port_clear_bits_raw(const struct device *dev, gpio_port_pins_t pins)
+{
+	const struct gpio_rz_config *config = dev->config;
+	struct gpio_rz_data *data = dev->data;
+	ioport_size_t mask = (ioport_size_t)pins;
+	ioport_size_t value = 0x00;
+	fsp_err_t err;
+
+	err = config->fsp_api->portWrite(data->fsp_ctrl, config->fsp_port, value, mask);
+	if (err != FSP_SUCCESS) {
+		return -EIO;
+	}
+	return 0;
+}
+
+static int gpio_rz_port_toggle_bits(const struct device *dev, gpio_port_pins_t pins)
+{
+	const struct gpio_rz_config *config = dev->config;
+	struct gpio_rz_data *data = dev->data;
+	bsp_io_port_pin_t port_pin;
+	gpio_flags_t pre_flags;
+	ioport_size_t value = 0;
+	fsp_err_t err;
+
+	for (uint8_t idx = 0; idx < config->ngpios; idx++) {
+		if (pins & (1U << idx)) {
+			port_pin = config->fsp_port | idx;
+			gpio_rz_pin_config_get_raw(port_pin, &pre_flags);
+			if (pre_flags & GPIO_OUTPUT_INIT_HIGH) {
+				value &= (1U << idx);
+			} else if (pre_flags & GPIO_OUTPUT_INIT_LOW) {
+				value |= (1U << idx);
+			}
+		}
+	}
+	err = config->fsp_api->portWrite(data->fsp_ctrl, config->fsp_port, value,
+					 (ioport_size_t)pins);
+	if (err != FSP_SUCCESS) {
+		return -EIO;
+	}
+	return 0;
+}
+
+#define GPIO_RZ_HAS_INTERRUPT DT_HAS_COMPAT_STATUS_OKAY(renesas_rz_gpio_int)
+
+#if GPIO_RZ_HAS_INTERRUPT
+static int gpio_rz_int_disable(const struct device *dev, uint8_t tint_num)
+{
+	struct gpio_rz_tint_data *data = dev->data;
+	volatile uint32_t *tssr = &R_INTC_IM33->TSSR0;
+	volatile uint32_t *titsr = &R_INTC_IM33->TITSR0;
+	volatile uint32_t *tscr = &R_INTC_IM33->TSCR;
+
+	tssr = &tssr[tint_num / 4];
+	titsr = &titsr[tint_num / 16];
+
+	irq_disable(GPIO_RZ_TINT_IRQ_GET(tint_num));
+	*tssr &= ~(0xFF << GPIO_RZ_TSSR_OFFSET(tint_num));
+	*titsr &= ~(0x3 << GPIO_RZ_TITSR_OFFSET(tint_num));
```

Comment:
Can we explain this mask? Either through a define or at least a dedicated comment

Reply:
Thank you for the feedback.
This mask is used to get to position of the bit related to interrupt number in the register and clear it to default value.
I also added the comment it.
Please kindly check it.